### PR TITLE
Added ability to parse Redshift DISTKEY, SORTKEY, DISTSTYLE

### PR DESCRIPTION
**Pull request by yuchaofan13**
Example syntax: CREATE TABLE SOUP DISTKEY(soup1) SORTKEY (soup2) DISTSTYLE AUTO

For now I've created individual properties for each and added these to ROOT_PROPERTIES in the base generator. However, distkey and sortkey both use parentheses- not sure where I should be putting their properties instead. Also not sure whether I should have something like 'allow_distkey = True' in the Redshift dialect subclass.

### CODE REVIEW

File: sqlglot/generator.py [hunk 1]
```diff
@@ -94,6 +94,9 @@ class Generator:
     ROOT_PROPERTIES = {
         exp.ReturnsProperty,
         exp.LanguageProperty,
+        exp.DistStyleProperty,
+        exp.DistKeyProperty,
+        exp.SortKeyProperty,
```
Comment (GeorgeSittas) [inline sqlglot/generator.py:99]: Should we override `ROOT_PROPERTIES` for Redshift and add them there?

File: sqlglot/parser.py [hunk 2]
```diff
@@ -452,6 +456,9 @@ class Parser(metaclass=_Parser):
         TokenType.PARTITIONED_BY: lambda self: self._parse_partitioned_by(),
         TokenType.SCHEMA_COMMENT: lambda self: self._parse_schema_comment(),
         TokenType.STORED: lambda self: self._parse_stored(),
+        TokenType.DISTKEY: lambda self: self._parse_distkey(),
+        TokenType.DISTSTYLE: lambda self: self._parse_diststyle(),
+        TokenType.SORTKEY: lambda self: self._parse_sortkey(),
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:461]: Same here: should we add these to Redshift only?

File: sqlglot/parser.py [hunk 3]
```diff
@@ -846,6 +857,37 @@ def _parse_stored(self):
...
+        )
+
+    def _parse_diststyle(self):
+        this = exp.Literal.string("DISTSTYLE")
+        value = exp.Literal.string(self._parse_var().name)
+        return self.expression(
+            exp.DistStyleProperty,
+            this=this,
+            value=value,
+        )
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:889]: Same

**Review by GeorgeSittas — COMMENTED** [review]
I left a few comments.

By the way, I think the distkey/sortkey properties you can have a compound key. This is documented [here](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html), so we might want to change how we parse / represent these two?

File: sqlglot/generator.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/generator.py:99]: yea let's add these to redshift

File: sqlglot/parser.py (hunk 2 above)
Comment (tobymao) [reply sqlglot/parser.py:461]: probably ok to be here

**Comment by GeorgeSittas** [comment]
@yuchaofan13 regarding what I said earlier:

> By the way, I think the distkey/sortkey properties can have a compound key. This is documented [here](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html), so we might want to change how we parse / represent these two?

It's ok if we push your changes for now. I'll try to refactor the way properties are handled and I can look into that too.

**Comment by yuchaofan13** [comment]
@GeorgeSittas nice, thank you! My understanding of the codebase is still quite lacking and I wasn't sure how to modify the properties so will be curious to see the refactoring! (this is my first ever pull request hahaha) I hadn't actually heard of the compound key before but I have some other features in mind too.

**Comment by GeorgeSittas** [comment]
> so will be curious to see the refactoring

@yuchaofan13 I've opened a PR about this: https://github.com/tobymao/sqlglot/pull/733.

**Pull request merged**